frontend/src/reducers/discover.js

```diff
@@ -0,0 +1,27 @@
+import merge from 'lodash/object/merge';
+import { DISCOVER_SUCCESS } from '../constants/discover';
+
+export default function discover(state={}, action={}) {
+  switch (action.type) {
+    case DISCOVER_SUCCESS:
+    const discover = action.json;
+    if (state.show) {
```

Comment:
I will explain my thought

`if (state.show === discover.show && state.sort === discover.sort) ...`
When this condition is `true` results are added to the `discover.collectives` array.
This is the only case when paginating via scrolling down.
Otherwise,`discover.collectives` array is replaced with the result (not appened).

If the first condition is `true`, the there is another condition inside:
`if (state.collectives) { ...`
This test is simply to separate the case where `state.collectives` is undefined, which is only true once, every time the page loads.
If `state.collectives` exists, then we are certain that the user is pagination/scrolling.

I will explain the `unshift`.
```
          for (let i = state.collectives.length - 1; i >= 0; i--) {
            discover.collectives.unshift(state.collectives[i]);
          }
```
The current **state** results found in `state.collectives` are prepended to the new results in `discover.collectives`.
This is done so the older or first results come first in the array `discover.collectives` which becomes the new **state** results.

I suppose it simply came naturally to me to do write it this way, I realize now how it can be a bit cryptic. The problem I solved by taking this approach was that when change filters: either  `Show` or `Sort`, old result requested with different filter settings would not be removed from the `state.discover.collectives` array (which is what is rendered), instead the x number of new results would **overwrite** the older results, up-to the size of new results.

**I will explain:**  We have 5 old results, we change the filters and get 3 new results. Out of the 5 old result, the first 3 will be replaced with the 3 new results, the last 2 old results will not be removed or altered.

This happened when using `return merge({}, state, discover);`